diff --git a/xbmc/settings/Settings.cpp b/xbmc/settings/Settings.cpp
index ecd155d..9aaf6ed 100644
--- a/xbmc/settings/Settings.cpp
+++ b/xbmc/settings/Settings.cpp
@@ -76,6 +76,7 @@
 #include "threads/SingleLock.h"
 #include "utils/CharsetConverter.h"
 #include "utils/log.h"
+#include "utils/md5.h"
 #include "utils/RssManager.h"
 #include "utils/StringUtils.h"
 #include "utils/SystemInfo.h"
@@ -529,6 +530,17 @@ bool CSettings::Save(const std::string &file)
   if (!m_settingsManager->Save(root))
     return false;
 
+  // Avoid saving if the settings saved earlier are indetical to the current ones
+  if (CFile::Exists(file))
+  {
+    std::string fileMD5 = CUtil::GetFileMD5(file);
+    TiXmlPrinter xmlPrinter;
+    xmlDoc.Accept(&xmlPrinter);
+    std::string settingsMD5 = XBMC::XBMC_MD5::GetMD5(xmlPrinter.CStr());
+    if (fileMD5 == settingsMD5)
+      return true;
+  }
+
   return xmlDoc.SaveFile(file);
 }
 
